pkgs/tools/package-management/libdnf/default.nix

```diff
@@ -0,0 +1,59 @@
+{ gcc9Stdenv, stdenv, fetchFromGitHub, cmake, gettext, pkgconfig, gpgme, libsolv, openssl, check, pcre, json_c, libmodulemd, libsmartcols, sqlite, librepo, libyaml, rpm }:
+
+gcc9Stdenv.mkDerivation rec {
+  pname = "libdnf";
+  version = "0.55.2";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0hiydwfa90nsrqk5ffa6ks1g73wnsgjgq7z7gwq9jj76a7gpfbfq";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    pkgconfig
+    check
+  ];
+
+  propagatedBuildInputs = [
+    gpgme
+    openssl
+    json_c
+    sqlite
+    librepo
+    pcre
+    rpm
+    libsolv
+    libsmartcols
+    libyaml
+    libmodulemd
+  ];
+
+  patches = stdenv.lib.optionals stdenv.isDarwin [ ./darwin.patch ];
```

Comment:
Please apply patches everywhere if they don't break other platforms.

Reply:
I think this one does break linux.